File: app/src/main/java/net/osomi/phomera/FullscreenActivity.java
package net.osomi.phomera;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;
import android.widget.TextView;

import org.freedesktop.gstreamer.GStreamer;

public class FullscreenActivity extends AppCompatActivity {

    @Override
    public void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);

        try {
            GStreamer.init(this);
        }
        catch (Exception e) {
            Toast.makeText(this, e.getMessage(), Toast.LENGTH_LONG).show();
            finish();
            return;
        }

        getSupportActionBar().hide();
        setContentView(R.layout.activity_fullscreen);

        findViewById(R.id.div_bottom).setVisibility(View.VISIBLE);
        findViewById(R.id.div_extended).setVisibility(View.GONE);

        findViewById(R.id.more).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                toggleMore();
            }
        });

        ((TextView)findViewById(R.id.version)).setText(getGStreamerVersion());
    }

    private void toggleMore() {

        if(findViewById(R.id.div_extended).getVisibility() == View.GONE)
            findViewById(R.id.div_extended).setVisibility(View.VISIBLE);

        else findViewById(R.id.div_extended).setVisibility(View.GONE);
    }

    public native String getGStreamerVersion();

    static {
        System.loadLibrary("gstreamer_android");
        System.loadLibrary("phomera");
    }
}

File: app/src/main/java/net/osomi/phomera/FullscreenActivity.kt
package net.osomi.phomera
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.view.View
import kotlinx.android.synthetic.main.activity_fullscreen.*
import androidx.core.app.ComponentActivity
import androidx.core.app.ComponentActivity.ExtraData
import androidx.core.content.ContextCompat.getSystemService
import android.icu.lang.UCharacter.GraphemeClusterBreak.T
import android.view.Window
import android.view.WindowManager

class FullscreenActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)

        supportActionBar?.hide()
        setContentView(R.layout.activity_fullscreen)

        div_bottom.visibility = View.VISIBLE
        div_extended.visibility = View.GONE;

        more.setOnClickListener { toggleMore() }

        version.text = version.text.toString() + getGStreamerVersion();
    }

    private fun toggleMore() {

        if(div_extended.visibility == View.GONE)
            div_extended.visibility = View.VISIBLE;

        else div_extended.visibility = View.GONE;
    }

    external fun getGStreamerVersion(): String

    companion object {
        init {
            // System.loadLibrary("gstreamer_android")
            System.loadLibrary("phomera")
        }
    }
}
